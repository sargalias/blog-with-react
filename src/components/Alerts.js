import React from 'react';

const Alerts = ({ alerts }) => {
  if (alerts)
    return (
      <section id="alerts">
        <div className="container">
          { alerts.map(({ type, text }) => (
            <div className={`alert alert-${type} alert-dismissible fade show`} role="alert">
              {text}
              <button type="button" className="close" data-dismiss="alert" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
          ))}
        </div>
      </section>
    );
};

export default Alerts;

