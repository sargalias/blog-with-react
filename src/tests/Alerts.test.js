import React from 'react';
import { shallow } from 'enzyme';
import Alerts from '../components/Alerts';

test('Alerts should not render without alerts prop', () => {
  const wrapper = shallow(<Alerts />);
  expect(wrapper.html()).toBeNull();
});

test('Alerts should render correctly with one alert', () => {
  const alerts = [
    { type: 'danger', text: 'test' }
  ];
  const wrapper = shallow(<Alerts alerts={alerts} />);
  expect(wrapper).toMatchSnapshot();
});

test('Alerts should render correctly with two alerts', () => {
  const alerts = [
    { type: 'danger', text: 'test' },
    { type: 'success', text: 'test 2' }
  ];
  const wrapper = shallow(<Alerts alerts={alerts} />);
  expect(wrapper).toMatchSnapshot();
});

test('Alerts should call Alert correctly with one alert', () => {
  const alerts = [
    { type: 'danger', text: 'test' }
  ];
  const wrapper = shallow(<Alerts alerts={alerts} />);
  const alert = wrapper.find('Alert');
  expect(alert.props()).toEqual(alerts[0]);
});

test('Alerts should call Alert correctly with two alerts', () => {
  const alerts = [
    { type: 'danger', text: 'test' },
    { type: 'success', text: 'test2' }
  ];
  const wrapper = shallow(<Alerts alerts={alerts} />);
  const alertComponents = wrapper.find('Alert');
  alertComponents.forEach((alert, i) => {
    const props = alert.props();
    expect(props).toEqual(alerts[i]);
  });
});
