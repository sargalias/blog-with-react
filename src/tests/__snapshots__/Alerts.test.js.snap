// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts should render correctly with one alert 1`] = `
<section
  id="alerts"
>
  <div
    className="container"
  >
    <div
      className="alert alert-danger alert-dismissible fade show"
      role="alert"
    >
      test
      <button
        aria-label="Close"
        className="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`Alerts should render correctly with two alerts 1`] = `
<section
  id="alerts"
>
  <div
    className="container"
  >
    <div
      className="alert alert-danger alert-dismissible fade show"
      role="alert"
    >
      test
      <button
        aria-label="Close"
        className="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
    <div
      className="alert alert-success alert-dismissible fade show"
      role="alert"
    >
      test 2
      <button
        aria-label="Close"
        className="close"
        data-dismiss="alert"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
      </button>
    </div>
  </div>
</section>
`;
